src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -107,94 +92,55 @@ private void OnSolutionCrawlerProgressChanged(object sender, ProgressData progre
-                {
-                    _resettableDelay.Reset();
-                    return;
-                }
-
-                _resettableDelay = new ResettableDelay((int)s_minimumInterval.TotalMilliseconds, AsynchronousOperationListenerProvider.NullListener);
-                _resettableDelay.Task.SafeContinueWith(_ =>
-                {
-                    _resettableDelay = null;
-                    ProgressUpdated();
-                }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
-            }
         }
 
-        private void StartedOrStopped(bool started)
+        private void Started()
         {
-            if (started)
-            {
-                ResetProgressStatus();
+            ResetProgressStatus();
 
-                // if there is any pending one. make sure it is finished.
-                _currentTask?.TrySetResult(default);
+            // if there is any pending one. make sure it is finished.
+            _currentTask?.TrySetResult(default);
 
-                var taskHandler = _taskCenterService.PreRegister(_options, data: default);
+            var taskHandler = _taskCenterService.PreRegister(_options, data: default);
 
-                _currentTask = new TaskCompletionSource<VoidResult>();
-                taskHandler.RegisterTask(_currentTask.Task);
+            _currentTask = new TaskCompletionSource<VoidResult>();
+            taskHandler.RegisterTask(_currentTask.Task);
 
-                // report initial progress
-                ChangeProgress(taskHandler, message: null);
+            // report initial progress
+            ChangeProgress(taskHandler, message: null);
 
-                // set handler
-                _taskHandler = taskHandler;
-            }
-            else
-            {
-                // clear progress message
-                ChangeProgress(_taskHandler, message: null);
+            // set handler
+            _taskHandler = taskHandler;
+        }
 
-                // stop progress
-                _currentTask?.TrySetResult(default);
-                _currentTask = null;
+        private void Stopped()
+        {
+            // clear progress message
+            ChangeProgress(_taskHandler, message: null);
 
-                _taskHandler = null;
+            // stop progress
+            _currentTask?.TrySetResult(default);
+            _currentTask = null;
 
-                ResetProgressStatus();
-            }
+            _taskHandler = null;
+
+            ResetProgressStatus();
         }
 
         private void ResetProgressStatus()
         {
             _lastPendingItemCount = 0;
             _lastProgressStatus = ProgressStatus.Paused;
-            _lastShownProgressStatus = ProgressStatus.Paused;
         }
 
-        private static void ChangeProgress(ITaskHandler taskHandler, string message)
+        private static void ChangeProgress(ITaskHandler? taskHandler, string? message)
```

Comment:
how would we not have a task handler?  why are we even doing *anything* in that case?